design/pipeline-resources.md

```diff
@@ -27,73 +27,63 @@ resources:
 
 If you have a pipeline that produces artifacts, you can consume the artifacts by defining a `pipelines` resource. A pipeline can be another Azure DevOps pipeline or any external pipelines like Jenkins etc.
 
-
 ### Schema
 
 ```yaml
 resources:        # types: pipelines | repositories | containers | packages
   pipelines:
   - pipeline: string  # identifier for the pipeline resource
-    type: enum  # type of the pipeline source like azurePipelines, Jenkins etc. 
-    connection: string  # service connection to connect to the source
-    project: string # project for the source; optional for current project  
+    connection: string  # service connection for pipelines in other Azure DevOps organizations or Servers
+    project: string # project for the source; optional for current project
```

Comment:
Have you considered moving branch and tags under versionSpec? Something like
```yaml
- pipeline: alias
  project: ""
  source: ""
  versionSpec:
    quality: succeeded
    ref: master
    tags: ""
```

If not, what are the values for `versionSpec` other than `Latest`?

Consider `ref` instead of `branch` (match repository)

Is `tags` a sequence? Or is it a string? If it is a sequence, are they AND'd?

How are you envisioning adding trigger specifications? With a `trigger` property, and then everything relevant nested?

Reply:
Leaving this for @ashokirla as well.

Reply:
We initially started with that design by keeping branch, tags, version under one node. I discussed this with Chris and Matt. Since these are anyway pipeline resource specific properties, the additional depth is not required. We can achieve all the scenarios by having version, branch and tag inputs at the same level. 

Regarding 'branch' vs 'ref', a pipeline can be associated with any kind of version control system. So we think keeping it generic as 'branch' would be more perceivable to user :).

Tag is a sequence i.e. list of strings. If multiple tags are given, they are AND'd. 

Triggers will be a separate property in pipeline resource and all the trigger specific properties will be nested under that node. I will share a separate spec for Triggers.

We will consider latest successfully completed pipeline run. This means all the stages of the pipeline should be succeeded or partially succeeded.  Like I explained above, we will start with this and then based on the scenarios we can extend.